packages/libs/deer/desert/src/token.rs

```diff
@@ -0,0 +1,75 @@
+use core::fmt::{Debug, Display, Formatter};
+
+use deer::Number;
+
+// TODO: test
+#[derive(Debug, Copy, Clone)]
+pub enum Token {
+    /// A serialized `bool`
+    ///
+    /// ```
+    /// # use error_stack::ResultExt;
+    /// use deer::{
+    ///     error::{DeserializeError, VisitorError},
+    ///     Deserialize, Deserializer, Document, Reflection, Schema, Visitor,
+    /// };
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// #[derive(Debug, PartialEq)]
+    /// struct Bool(bool);
+    ///
+    /// impl Reflection for Bool {
+    ///     fn schema(_: &mut Document) -> Schema {
+    ///         Schema::new("boolean")
+    ///     }
+    /// }
+    ///
+    /// impl<'de> Deserialize<'de> for Bool {
+    ///     type Reflection = Self;
+    ///
+    ///     fn deserialize<D: Deserializer<'de>>(de: D) -> error_stack::Result<Self, DeserializeError> {
+    ///         struct BoolVisitor;
+    ///
+    ///         impl<'de> Visitor<'de> for BoolVisitor {
+    ///             type Value = Bool;
+    ///
+    ///             fn expecting(&self) -> Document {
+    ///                 Bool::reflection()
+    ///             }
+    ///
+    ///             fn visit_bool(self, v: bool) -> error_stack::Result<Self::Value, VisitorError> {
+    ///                 Ok(Bool(v))
+    ///             }
+    ///         }
+    ///
+    ///         de.deserialize_bool(BoolVisitor)
+    ///             .change_context(DeserializeError)
+    ///     }
+    /// }
+    ///
+    /// assert_tokens(&Bool(true), &[Token::Bool(true)])
+    /// ```
+    Bool(bool),
```

Comment:
This is more of a showcase. Building this out, I realized that most of the code is duplicated for these kinds of tests on the primitive types, so the idea was to showcase the examples instead of using the existing primitive `Visitor` and `Deserialize` implementations that are in the follow-up PR.